tests/framework_django/test_application.py

```diff
@@ -12,297 +12,259 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from testing_support.fixtures import (
-    override_application_settings,
-    override_generic_settings, override_ignore_status_codes)
-from testing_support.validators.validate_code_level_metrics import validate_code_level_metrics
-from newrelic.hooks.framework_django import django_settings
-from testing_support.validators.validate_transaction_metrics import validate_transaction_metrics
-from testing_support.validators.validate_transaction_errors import validate_transaction_errors
-
 import os
 
 import django
+from testing_support.fixtures import (
+    override_application_settings,
+    override_generic_settings,
+    override_ignore_status_codes,
+)
+from testing_support.validators.validate_code_level_metrics import (
+    validate_code_level_metrics,
+)
+from testing_support.validators.validate_transaction_errors import (
+    validate_transaction_errors,
+)
+from testing_support.validators.validate_transaction_metrics import (
+    validate_transaction_metrics,
+)
+
+from newrelic.hooks.framework_django import django_settings
 
-DJANGO_VERSION = tuple(map(int, django.get_version().split('.')[:2]))
-DJANGO_SETTINGS_MODULE = os.environ.get('DJANGO_SETTINGS_MODULE', None)
+DJANGO_VERSION = tuple(map(int, django.get_version().split(".")[:2]))
+DJANGO_SETTINGS_MODULE = os.environ.get("DJANGO_SETTINGS_MODULE", None)
 
 
 def target_application():
     from _target_application import _target_application
+
     return _target_application
 
 
 # The middleware scoped metrics are dependent on the MIDDLEWARE_CLASSES or
 # MIDDLEWARE defined in the version-specific Django settings.py file.
 
 _test_django_pre_1_10_middleware_scoped_metrics = [
-    (('Function/django.middleware.common:'
-            'CommonMiddleware.process_request'), 1),
-    (('Function/django.contrib.sessions.middleware:'
-            'SessionMiddleware.process_request'), 1),
-    (('Function/django.contrib.auth.middleware:'
-            'AuthenticationMiddleware.process_request'), 1),
-    (('Function/django.contrib.messages.middleware:'
-            'MessageMiddleware.process_request'), 1),
-    (('Function/django.middleware.csrf:'
-            'CsrfViewMiddleware.process_view'), 1),
-    (('Function/django.contrib.messages.middleware:'
-            'MessageMiddleware.process_response'), 1),
-    (('Function/django.middleware.csrf:'
-            'CsrfViewMiddleware.process_response'), 1),
-    (('Function/django.contrib.sessions.middleware:'
-            'SessionMiddleware.process_response'), 1),
-    (('Function/django.middleware.common:'
-            'CommonMiddleware.process_response'), 1),
-    (('Function/django.middleware.gzip:'
-            'GZipMiddleware.process_response'), 1),
-    (('Function/newrelic.hooks.framework_django:'
-            'browser_timing_insertion'), 1),
+    (("Function/django.middleware.common:" "CommonMiddleware.process_request"), None),
```

Comment:
Is there an advantage to leaving the metrics in for the ignored middlewares with a count of `None` vs dropping them from the scoped metric list?

Reply:
I did this specifically to make sure they were dropped because if you simply don't look for them they could still be there but the test wouldn't catch it.